nmigen_boards/quickfeather.py

```diff
@@ -0,0 +1,86 @@
+import os
+import sys
+import subprocess
+
+from nmigen.build import *
+from nmigen.vendor.quicklogic import *
+from nmigen_boards.resources import *
+
+
+__all__ = ["QuickfeatherPlatform"]
+
+
+class QuickfeatherPlatform(QuicklogicPlatform):
+    device      = "ql-eos-s3_wlcsp"
+    part        = "PU64"
+    default_clk = "sys_clk0"
+    connectors = [
+        Connector("J", 2, "- 28 22 21 37 36 42 40 7 2 4 5"),
+        Connector("J", 3, "- 8 9 17 16 20 6 55 31 25 47 - - - - 41"),
+        Connector("J", 8, "27 26 33 32 23 57 56 3 64 62 63 61 59 - - -"),
+    ]
+    resources   = [
+        # This is a placeholder resource since the board utilizes
+        # default internal SoC clock.
+        Resource("sys_clk0", 0, Pins("63", dir="i"), Clock(10e6)),
+
+        *ButtonResources(pins="62"),
+
+        RGBLEDResource(0, r="34", g="39", b="38"),
+
+        UARTResource(0,
+            rx="9", tx="8",
+        ),
+
+        SPIResource(0,
+            cs="11", clk="20", copi="16", cipo="17"
+        ),
+        SPIResource(1,
+            cs="37", clk="40", copi="36", cipo="42",
+            role="peripheral"
+        ),
+
+        Resource("i2c", 0,
+            Subsignal("scl", Pins("4", dir="io")),
+            Subsignal("sda", Pins("5", dir="io")),
```

Comment:
Please use `I2CResource` here. (It wasn't upstream at the time you sent the PR--unfortunately so, or it would have prevented the issue with pin directions--so you might need to rebase.)